### PR TITLE
X and Y attributes for tabulated function.

### LINKED ISSUE #14642: Tabulated Function Requires Named Workspace
The [TabulatedFunction](http://docs.mantidproject.org/nightly/fitfunctions/TabulatedFunction.html) takes a Workspace via a name. This means that in order to run `Fit` as a child algorithm in a `DataProcessorAlgorithm`, one must perform an ADS hack to get that function to work. It would be much better from a usage perspective if it was not a requirement that the workspace sat in the ADS prior to the algorithm execution. It also leads to the potential of a runtime issues based on naming clashes while `DataProcessorAlgorithms` are executing simultaneously.

One solution could be to pass a Workspace to fit for the purposes of using the TabulatedFunction outside of the ADS. I think the solution should involve passing the workspace as a strongly typed object.

### PR DESCRIPTION
**Pull request by mantid-roman**
Fixes #14642.

New X and Y vector attributes is an additional source of data for the TabulatedFunction.
To test try using the function with the new attributes.

### CODE REVIEW

**Comment by FedeMPouzols** [comment]
I've restarted the windows and rhel7+system_tests builds which had failed some days ago.

This seems to produce an error in PythonAlgorithms_BayesQuasiTest, but curiously only on windows:
```
09:52:25 ======================================================================
09:52:25 FAIL: test_QLr_Run (__main__.BayesQuasiTest)
09:52:25 ----------------------------------------------------------------------
09:52:25 Traceback (most recent call last):
09:52:25   File "C:/Jenkins/workspace/pull_requests-win7/Framework/PythonInterface/test/python/plugins/algorithms/BayesQuasiTest.py", line 159, in test_QLr_Run
09:52:25     self._validate_Qlr_value(result, prob, fit_group)
09:52:25   File "C:/Jenkins/workspace/pull_requests-win7/Framework/PythonInterface/test/python/plugins/algorithms/BayesQuasiTest.py", line 76, in _validate_Qlr_value
09:52:25     self.assertEquals(round(probability.dataY(1)[0], 3), -375.123)
09:52:25 AssertionError: -375.124 != -375.123
09:52:25 
09:52:25 ======================================================================
09:52:25 FAIL: test_QSe_Run (__main__.BayesQuasiTest)
09:52:25 ----------------------------------------------------------------------
09:52:25 Traceback (most recent call last):
09:52:25   File "C:/Jenkins/workspace/pull_requests-win7/Framework/PythonInterface/test/python/plugins/algorithms/BayesQuasiTest.py", line 182, in test_QSe_Run
09:52:25     self._validate_QSe_value(result, fit_group)
09:52:25   File "C:/Jenkins/workspace/pull_requests-win7/Framework/PythonInterface/test/python/plugins/algorithms/BayesQuasiTest.py", line 127, in _validate_QSe_value
09:52:25     self.assertEquals(round(result.dataY(0)[0], 5), 8.2741)
09:52:25 AssertionError: 8.28044 != 8.2741
09:52:25 
```

**Comment by FedeMPouzols** [comment]
Ah, ok, this test effectively runs on windows only. Probably it just needs more updates, to these two expected values?

**Comment by FedeMPouzols** [comment]
I think it would be worth updating the documentation of TabulatedFunction to give an example with this new alternative way of passing the data (x, y vectors). The current example, which loads a workspace from a nexus file looks good but I'd add a comment/warning to users, as discussed in the issue, that using a workspace (by name) can have issues.

**Comment by martyngigg** [comment]
Is this required for `3.7`?

**Comment by NickDraper** [comment]
@mantid-roman is this needed for R3.7 or should it be 3.8?

Nothing has happened here since Feb

**Comment by martyngigg** [comment]
@mantid-roman I see this hasn't been touched in a while. Should it be pushed back again or just closed completely?

**Comment by mantid-roman** [comment]
It needs to be completed at some point.

**Comment by martyngigg** [comment]
Okay. Can we push it back  to `3.9`?

**Comment by martyngigg** [comment]
@mantid-roman There are conflicts to resolve here.

File: Framework/API/test/ExpressionTest.h [hunk 1]
```diff
@@ -367,6 +367,30 @@ class ExpressionTest : public CxxTest::TestSuite {
     TS_ASSERT_EQUALS(e.name(), ",");
   }
 
+  void testEmptyString() {
+    Expression e;
```
Comment (AnthonyLim23) [inline Framework/API/test/ExpressionTest.h]: I think that e is not a clear variable name.

**Comment by AnthonyLim23** [comment]
The tool works in python (see the script below), but the gui does not allow the X and Y values to be set (only the size of X and Y). When it is successful the spectra are as follows: 

- 0 is always zero across the whole x range.
- 1 is the X and Y values provided.
- 2 is the X and -Y values provided. 

the script is:
```

x = np.linspace(0, 10, 100)
y = x * 0
ws = CreateWorkspace(x, y)
EvaluateFunction('name=TabulatedFunction,X=(0, 5, 10),Y=(0,-5,6)', ws, OutputWorkspace='out')

```

**Comment by AnthonyLim23** [comment]
the gui now works. :shipit: once it has completed the build

**Comment by martyngigg** [comment]
Ubuntu tests are unrelated Vates failures.

**Pull request merged**